File: TOMIMSV4/Entity/Mantenimientos/Proveedor/clsBeProveedor_bodega.vb
Public Class clsBeProveedor_bodega
    Implements ICloneable
    Implements IDisposable

    Public Property IdAsignacion() As Integer = 0
    Public Property IdProveedor() As Integer = 0
    Public Property IdBodega() As Integer = 0
    Public Property Activo() As Boolean = False
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property IdAreaOrigen As Integer = 0

    Sub New()
    End Sub

    Sub New(ByRef IdAsignacion As Integer, ByVal IdProveedor As Integer, ByVal IdBodega As Integer, ByVal activo As Boolean, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date)
        Me.IdAsignacion = IdAsignacion
        Me.IdProveedor = IdProveedor
        Me.IdBodega = IdBodega
        Me.Activo = Activo
        Me.User_agr = User_agr
        Me.Fec_agr = Fec_agr
        Me.User_mod = User_mod
        Me.Fec_mod = Fec_mod
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function
End Class

File: WMS.EntityCore/Proveedor/clsBeProveedor_bodega.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMS.EntityCore.Proveedor
{
    public class clsBeProveedor_bodega : ICloneable
    {
        [Column("IdAsignacion")]
        [DisplayName("IdAsignacion")]
        public int IdAsignacion { get; set; } = 0;

        [Column("IdProveedor")]
        [DisplayName("IdProveedor")]
        public int IdProveedor { get; set; } = 0;

        [Column("IdBodega")]
        [DisplayName("IdBodega")]
        public int IdBodega { get; set; } = 0;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("IdAreaOrigen")]
        [DisplayName("IdAreaOrigen")]
        public int IdAreaOrigen { get; set; } = 0;
        public clsBeProveedor Proveedor { get; set; } = new clsBeProveedor();

        public clsBeProveedor_bodega() { }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}
